networkit/cpp/randomization/CurveballImpl.h

```diff
@@ -0,0 +1,245 @@
+/*
+ * CurveballImpl.h
+ *
+ * Author: Hung Tran <htran@ae.cs.uni-frankfurt.de>
+ */
+#ifndef RANDOMIZATION_CURVEBALL_IMPL_H
+#define RANDOMIZATION_CURVEBALL_IMPL_H
+
+#include <cassert>
+#include <utility>
+
+#include "../Globals.h"
+#include "../graph/Graph.h"
+
+namespace NetworKit {
+namespace CurveballDetails {
+
+// Global Definitions
+using TradeDescriptor = std::pair<node, node>;
+using tradeid_t = node;
+using count = node;
+
+using trade_vector = std::vector<TradeDescriptor>;
+using nodepair_vector = std::vector <std::pair<node, node> >;
+
+constexpr node INVALID_NODE = std::numeric_limits<node>::max();
+constexpr count LISTROW_END = std::numeric_limits<count>::max();
+constexpr tradeid_t TRADELIST_END = std::numeric_limits<tradeid_t>::max();
+
+struct edge_t : public std::pair<node, node> {
```

Comment:
I do not like the _t suffixes. node and count do not have suffices. As naming the type just `edge` might not be a good idea, I suggest to rename it. On the other hand, this struct adds little over std::pair<node, node>.